src/v/cloud_storage/remote.cc

```diff
@@ -779,22 +779,27 @@ void auth_refresh_bg_op::do_start_auth_refresh_op(
     } else {
         // Create an implementation of refresh_credentials based on the setting
         // cloud_credentials_source.
-        _refresh_credentials.emplace(cloud_roles::make_refresh_credentials(
-          _cloud_credentials_source,
-          _gate,
-          _as,
-          std::move(credentials_update_cb),
-          _region_name));
-
-        vlog(
-          cst_log.info,
-          "created credentials refresh implementation based on credentials "
-          "source "
-          "{}: {}",
-          _cloud_credentials_source,
-          *_refresh_credentials);
+        try {
+            _refresh_credentials.emplace(cloud_roles::make_refresh_credentials(
+              _cloud_credentials_source,
+              _gate,
+              _as,
+              std::move(credentials_update_cb),
+              _region_name));
 
-        _refresh_credentials->start();
+            vlog(
+              cst_log.info,
+              "created credentials refresh implementation based on credentials "
+              "source {}: {}",
+              _cloud_credentials_source,
+              *_refresh_credentials);
+            _refresh_credentials->start();
+        } catch (const std::exception& ex) {
+            vlog(
```

Comment:
The commit message says "so that the application terminates with proper logs", but this block is preventing the exception from propagating to the caller.

I think the behavior is probably right (redpanda should not terminate on credentials issues), so maybe the commit message needs updating.

Reply:
Yes that's right, my commit was badly worded, before this change a problem in the background op caused the application to crash ( #5507 ) and this change logs the exception and moves on. I have reworded the commit.